### PR TITLE
Add installation instructions

### PR DESCRIPTION
**Pull request by jeddy3**
and other docs revisions.

I had the opportunity to look through all the docs in one sitting with a fresh pair of eyes last week. Here’s some amends. Each is a separate commit to make the review easier.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -16,28 +16,28 @@ A mighty, modern CSS linter that helps you enforce consistent conventions and av
...
+2. Create your [configuration object](/docs/user-guide/configuration.md) by either extending a shared config or crafting your own one:
+    - To extend a shared config, we recommend using [`stylelint-config-standard`](https://github.com/stylelint/stylelint-config-standard). It includes over 80 of stylelint's rules with sensible defaults. (You can always override specific rules after extending the config.) We update the config with each new release of stylelint. Alternately, you can [search for](https://www.npmjs.com/browse/keyword/stylelint-config) a community config and [extend](/docs/user-guide/configuration.md#extends) that instead.
```
Comment (davidtheclark) [inline README.md]: I think this first sentence now downplays other shared configs too much. I prefer starting with

> If you want to use a pre-written config, [find one that you like](https://www.npmjs.com/browse/keyword/stylelint-config) and [extend it](/docs/user-guide/configuration.md#extends). We recommend trying [`stylelint-config-standard`](https://github.com/stylelint/stylelint-config-standard). It ...

File: README.md [hunk 2]
```diff
@@ -16,28 +16,28 @@ A mighty, modern CSS linter that helps you enforce consistent conventions and av
...
-    - If you want to use a pre-written config, just [find one](https://www.npmjs.com/browse/keyword/stylelint-config) and [extend it](/docs/user-guide/configuration.md#extends). We recommend trying [`stylelint-config-standard`](https://github.com/stylelint/stylelint-config-standard), which includes over 80 of stylelint's rules with sensible defaults. (You can always override specific rules after extending a config.)
-    - To craft your own from the ground up, [learn about rules](/docs/user-guide/about-rules.md). *All of the rules are off by default*, so you only have to learn about [the rules](/docs/user-guide/rules.md) you want to turn on and enforce. That way you can start small, growing your config over time as you have a chance to explore more of the rules. Alternately, copy-paste [this example configuration](/docs/user-guide/example-config.md), which lists all of stylelint's rules and their primary options, then remove (or turn off) the rules you don't want and edit the primary option of each rule to your liking.
+2. Create your [configuration object](/docs/user-guide/configuration.md) by either extending a shared config or crafting your own one:
```
Comment (davidtheclark) [inline README.md]: Scratch "one": `crafting your own:`

File: README.md [hunk 3]
```diff
@@ -16,28 +16,28 @@ A mighty, modern CSS linter that helps you enforce consistent conventions and av
...
+    - To extend a shared config, we recommend using [`stylelint-config-standard`](https://github.com/stylelint/stylelint-config-standard). It includes over 80 of stylelint's rules with sensible defaults. (You can always override specific rules after extending the config.) We update the config with each new release of stylelint. Alternately, you can [search for](https://www.npmjs.com/browse/keyword/stylelint-config) a community config and [extend](/docs/user-guide/configuration.md#extends) that instead.
+    - To craft your own from the ground up, [learn about how rules are named and how they work together](/docs/user-guide/about-rules.md). *All of the rules are off by default*, so you only have to learn about [the rules](/docs/user-guide/rules.md) you want to turn on and enforce. That way you can start small, growing your config over time as you have a chance to explore more of the rules. Alternately, copy-paste [this example configuration](/docs/user-guide/example-config.md), which lists all of stylelint's rules and their primary options, then remove (or turn off) the rules you don't want and edit the primary option of each rule to your liking.
```
Comment (davidtheclark) [inline README.md]: I think it would make sense, and break up this long paragraph, to separate the copy-paste method into its own bullet point. How about this:

> - Another approach is to copy-paste [this example configuration](/docs/user-guide/example-config.md), which lists all of stylelint's rules and their primary options. Then you can edit the options of each rule to your liking, and remove (or turn off with `null`) the rules that you don't care to enforce.

File: docs/developer-guide/rules.md [hunk 4]
```diff
@@ -59,9 +59,9 @@ Look at the messages of other rules to glean more conventions and patterns.
 
 *When writing the rule, always look to other similar rules for conventions and patterns to start from and mimic.*
 
-You will use the simple [PostCSS API](https://github.com/postcss/postcss/blob/master/docs/api.md) to navigate and analyze the CSS syntax tree.
+You will use the simple [PostCSS API](https://github.com/postcss/postcss/blob/master/docs/api.md) to navigate and analyze the CSS syntax tree. We recommend using the `walk` iterators (e.g. `walkDecls`), rather than using `foreach` to loop through the nodes.
```
Comment (davidtheclark) [inline docs/developer-guide/rules.md]: `forEach` camelcase

File: docs/developer-guide/rules.md [hunk 5]
```diff
@@ -59,9 +59,9 @@ Look at the messages of other rules to glean more conventions and patterns.
...
 
-Depending on the rule, we recommend using [postcss-value-parser](https://github.com/TrySound/postcss-value-parser) and/or [postcss-selector-parser](https://github.com/postcss/postcss-selector-parser). Use these standard parsers whenever possible, even if there's a performance hit, because there are significant gains in doing so.
+Depending on the rule, we also recommend using [postcss-value-parser](https://github.com/TrySound/postcss-value-parser) and/or [postcss-selector-parser](https://github.com/postcss/postcss-selector-parser). Use these standard parsers whenever possible, even if there's a performance hit, because there are significant gains in doing so.
```
Comment (davidtheclark) [inline docs/developer-guide/rules.md]: "and/or" could just be "and".

How about rewording the "Use these standard ..." sentence to:

> There are significant benefits to using these parsers instead of regular expressions or `indexOf` searches (even if they aren't always the most performant method).

File: docs/user-guide/cli.md [hunk 6]
```diff
@@ -1,10 +1,20 @@
 # The stylelint CLI
 
+## Installation
+
+stylelint is an [npm package](https://www.npmjs.com/package/stylelint). Install it using:
+
+```console
+npm install -g stylelint
```
Comment (davidtheclark) [inline docs/user-guide/cli.md]: I'd leave out the `-g`.

File: docs/user-guide/configuration.md [hunk 7]
```diff
@@ -18,7 +18,7 @@ The configuration search can be short-circuited by using either the `config` or
 
 ## The configuration object
 
-The configuration object can have the following properties. Only `rules` is required.
+The configuration object can have the following properties.
```
Comment (davidtheclark) [inline docs/user-guide/configuration.md]: Why cut that sentence about rules being required?

File: docs/user-guide/faq.md [hunk 8]
```diff
@@ -60,7 +54,7 @@ If you're using SUITCSS, you'll probably want to use [their shareable config](ht
 
 ## How do I manage conflicts between rules?
 
-Each rule stands alone, so sometimes it's possible to configure rules such that they conflict with one another. For example, you could configure the `string-quotes` rule to `"single"`, but configure `font-family-name-quotes` to use `"double-where-recommended"`. Or turn on two conflicting blacklist and whitelist rules e.g. `unit-blacklist` and `unit-whitelist`.
+Each rule stands alone, so sometimes it's possible to configure rules such that they conflict with one another. For example, you could turn on two conflicting blacklist and whitelist rules e.g. `unit-blacklist` and `unit-whitelist`.
```
Comment (davidtheclark) [inline docs/user-guide/faq.md]: Add a comma before `e.g.`

**Comment by davidtheclark** [comment]
Great work @jeddy3! I added a few suggestions.

File: README.md (hunk 1 above)
Comment (jeddy3) [inline README.md]: That was intentional. `config-standard` is the only config we maintain and, more importantly, release a new version of with each stylelint release. I think it's the best way of keeping up to date with error-catching rule releases i.e. by simply doing a `npm i stylelint-config-standard@latest` rather than looking through the changelog and doing it manually or relying on your community config maintainer to do it. It's also the one place where we [*contextually* list](https://github.com/stylelint/stylelint-config-standard#suggested-additions) all the other divisive rules - which makes extending it very easy. For these two reasons, I think it *is* different from community configs and should be emphasised accordingly.

File: README.md (hunk 2 above)
Comment (jeddy3) [inline README.md]: Sgtm

File: README.md (hunk 3 above)
Comment (jeddy3) [inline README.md]: Sgtm.

File: docs/developer-guide/rules.md (hunk 4 above)
Comment (jeddy3) [inline docs/developer-guide/rules.md]: Sgtm.

File: docs/developer-guide/rules.md (hunk 5 above)
Comment (jeddy3) [inline docs/developer-guide/rules.md]: Sgtm.

File: docs/user-guide/configuration.md (hunk 7 above)
Comment (jeddy3) [inline docs/user-guide/configuration.md]: It's not. You could have a valid config with just an `"extend"` in.

File: docs/user-guide/faq.md (hunk 8 above)
Comment (jeddy3) [inline docs/user-guide/faq.md]: Sgtm.

File: docs/user-guide/configuration.md (hunk 7 above)
Comment (davidtheclark) [inline docs/user-guide/configuration.md]: Ah, I forgot!

File: README.md (hunk 1 above)
Comment (davidtheclark) [inline README.md]: Ok!

**Comment by jeddy3** [comment]
@davidtheclark There's two new commits I'd like to get into this PR, if you don't mind reviewing them.

I've incorporate your feedback. When I went to break out the copy-paste "getting started" section I realised that learning about rule's names and how they work together is a prerequisite to both the "crafting" approaching i.e. choosing which primary options to use in that example config won't make much sense (especially for the whitespace rules) if you haven't read that document beforehand. So, I added them as sub-bullets. Can you think of a better way, or is that clear?

File: docs/developer-guide/rules.md [hunk 9]
```diff
@@ -28,9 +28,19 @@ We take care to ensure that all the rules are named accurately and consistently.
...
+Every rule *must have* a **primary option**.
+
+- In `"color-hex-case": "upper"`, the primary option is `"upper"`
+- In `"indentation": [2, { "except": ["block"] }]`, the primary option is `2`.
+
+#### Secondary
+
+Some rules require extra flexibility to address a variety of use-cases. These can use a **secondary options object**.
+
+- In `"color-hex-case": "upper"`, there is no secondary options object
```
Comment (davidtheclark) [inline docs/developer-guide/rules.md]: Missing a period.

**Comment by davidtheclark** [comment]
Fantastic.

File: docs/developer-guide/rules.md (hunk 9 above)
Comment (jeddy3) [inline docs/developer-guide/rules.md]: Added full stops in https://github.com/stylelint/stylelint/commit/81cfb916c6ba50d487dd6108d75c6565a030a9a7
